source/dub/packagemanager.d

```diff
@@ -60,6 +60,15 @@ public string toString (PlacementLocation loc) @safe pure nothrow @nogc
     }
 }
 
+/// A SelectionsFile associated with its file-system path.
+struct SelectionsFileLookupResult {
+	/// The absolute path to the dub.selections.json file
+	/// (potentially inherited from a parent directory of the root package).
+	NativePath absolutePath;
+	/// The parsed dub.selections.json file.
+	SelectionsFile selectionsFile;
+}
+
```

Comment:
I would like to fold this into `SelectionsFile`, but I know it's not currently possible. All things considered, it might be a good thing to add to `Configy`, so I'll have a look at it myself. It should not block this being merged.